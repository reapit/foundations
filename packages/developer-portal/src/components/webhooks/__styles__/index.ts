import { css } from '@linaria/core'
import { styled } from '@linaria/react'

export const IconContainer = styled.div`
  position: relative;
  height: 140px;

  svg {
    height: 140px;
  }
`

export const searchMinWidth = css`
  min-width: 210px;
`

export const ControlsContainer = styled.div`
  background-color: #fff;
  margin-bottom: 0.5rem;
`

export const inputFullWidth = css`
  input {
    width: 100%;
  }
`

export const overflowHidden = css`
  overflow: hidden;
`

export const DownloadButtonContainer = styled.div`
  padding: 1rem;
  background-color: var(--color-grey-100);
  border-radius: 0.25rem;
`

export const createCta = css`
  @media screen and (min-width: 1024px) {
    margin-left: 6rem;
  }
`
