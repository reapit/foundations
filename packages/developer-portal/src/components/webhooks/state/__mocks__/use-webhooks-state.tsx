import { mockAppSummaryModelPagedResult } from '../../../../tests/__stubs__/apps'
import { mockInstallationModelPagedResult } from '../../../../tests/__stubs__/installations'
import { mockTopicModelPagedResult } from '../../../../tests/__stubs__/webhooks'
import { defaultWebhooksFilterState } from '../defaults'

export const mockWebhooksState = {
  webhooksDataState: {
    apps: mockAppSummaryModelPagedResult,
    topics: mockTopicModelPagedResult,
    installations: mockInstallationModelPagedResult,
  },
  webhooksFilterState: defaultWebhooksFilterState,
}

export const useWebhooksState = jest.fn(() => mockWebhooksState)
