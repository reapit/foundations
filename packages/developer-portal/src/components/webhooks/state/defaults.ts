import dayjs from 'dayjs'
import { WebhooksFilterState } from './use-webhooks-state'

export const defaultWebhooksFilterState: WebhooksFilterState = {
  applicationId: '',
  from: dayjs().format('YYYY-MM-DD'),
  to: dayjs().format('YYYY-MM-DD'),
}
