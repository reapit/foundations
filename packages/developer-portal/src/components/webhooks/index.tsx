import React, { FC, ChangeEvent } from 'react'
import { NavigateFunction, Route, Routes, useNavigate } from 'react-router-dom'
import { Button, ButtonGroup, MobileControls, Tabs, Title, useMediaQuery, useModal } from '@reapit/elements'
import RoutePaths from '../../constants/routes'
import WebhooksAbout from './webhooks-about'
import WebhooksManage from './webhooks-manage'
import WebhooksLogs from './webhooks-logs'
import WebhooksNew from './webhooks-new'
import WebhooksControls from './webhooks-controls'
import { useWebhooksState } from './state/use-webhooks-state'

export const handleChangeTab = (navigate: NavigateFunction) => (event: ChangeEvent<HTMLInputElement>) => {
  navigate(`${event.target.value}${window.location.search}`)
}

export const WebhooksWrapper: FC = () => {
  const navigate = useNavigate()
  const { webhooksDataState } = useWebhooksState()
  const { Modal, openModal, closeModal } = useModal()
  const { isMobile } = useMediaQuery()
  const { apps } = webhooksDataState
  const { pathname } = window.location
  const isAboutPage = pathname === RoutePaths.WEBHOOKS_ABOUT
  const isManagePage = pathname === RoutePaths.WEBHOOKS_MANAGE
  const isLogsPage = pathname === RoutePaths.WEBHOOKS_LOGS
  const isNewPage = !isAboutPage && !isLogsPage && !isManagePage

  return (
    <>
      <Title>Webhooks</Title>
      <MobileControls onClick={openModal} />
      {isMobile && apps?.data && (
        <Modal title="Filters">
          <WebhooksControls />
          <ButtonGroup alignment="right">
            <Button intent="primary" onClick={closeModal}>
              Close
            </Button>
          </ButtonGroup>
        </Modal>
      )}
      <Tabs
        name="webhook-tabs"
        isFullWidth
        onChange={handleChangeTab(navigate)}
        options={[
          {
            id: 'webhook-tab-about',
            value: RoutePaths.WEBHOOKS_ABOUT,
            text: 'About Webhooks',
            isChecked: isAboutPage,
          },
          {
            id: 'webhook-tab-new',
            value: RoutePaths.WEBHOOKS_NEW,
            text: 'Add Webhook',
            isChecked: isNewPage,
          },
          {
            id: 'webhook-tab-manage',
            value: RoutePaths.WEBHOOKS_MANAGE,
            text: 'Manage Webhooks',
            isChecked: isManagePage,
          },
          {
            id: 'webhook-tab-logs',
            value: RoutePaths.WEBHOOKS_LOGS,
            text: 'Transaction Logs',
            isChecked: isLogsPage,
          },
        ]}
      />
      <Routes>
        <Route path={RoutePaths.WEBHOOKS_NEW.replace('/webhooks/', '')} element={<WebhooksNew />} />
        <Route path={RoutePaths.WEBHOOKS_MANAGE.replace('/webhooks/', '')} element={<WebhooksManage />} />
        <Route path={RoutePaths.WEBHOOKS_LOGS.replace('/webhooks/', '')} element={<WebhooksLogs />} />Ã¥
        <Route path={RoutePaths.WEBHOOKS_ABOUT.replace('/webhooks/', '')} element={<WebhooksAbout />} />
      </Routes>
    </>
  )
}

export default WebhooksWrapper
