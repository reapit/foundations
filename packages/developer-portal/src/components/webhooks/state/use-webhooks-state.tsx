import { Marketplace } from '@reapit/foundations-ts-definitions'
import React, {
  FC,
  createContext,
  useContext,
  useState,
  SetStateAction,
  Dispatch,
  useEffect,
  PropsWithChildren,
} from 'react'
import { useReapitConnect } from '@reapit/connect-session'
import { GetActionNames, getActions } from '@reapit/use-reapit-data'
import { useReapitGet } from '@reapit/use-reapit-data'
import { reapitConnectBrowserSession } from '../../../core/connect-session'
import { defaultWebhooksFilterState } from './defaults'
import { handleHistoryToQueryParams } from '../webhooks-controls'
import { TopicModel, TopicModelPagedResult } from '../../../types/webhooks'

export interface WebhooksDataState {
  apps: Marketplace.AppSummaryModelPagedResult | null
  topics: TopicModel[]
  installations: Marketplace.InstallationModelPagedResult | null
}

export interface WebhooksFilterState {
  applicationId: string
  from: string
  to: string
  entityId: string
  topicId: string
  eventId: string
}

export interface WebhooksStateHook {
  webhooksDataState: WebhooksDataState
  webhooksFilterState: WebhooksFilterState
  setWebhooksFilterState: Dispatch<SetStateAction<WebhooksFilterState>>
}

export const WebhooksStateContext = createContext<WebhooksStateHook>({} as WebhooksStateHook)

const { Provider } = WebhooksStateContext

export const handleSetTopics =
  (
    appDetail: Marketplace.AppDetailModel | null,
    allTopics: TopicModelPagedResult | null,
    setTopics: Dispatch<SetStateAction<TopicModel[]>>,
  ) =>
  () => {
    if (appDetail && allTopics?._embedded) {
      const filterdTopics = allTopics._embedded.filter((topic) => {
        if (!topic.associatedScope) return true
        return Boolean(appDetail.scopes?.find((scope) => scope.name === topic.associatedScope))
      })

      setTopics(filterdTopics)
    }
  }

export const WebhooksProvider: FC<PropsWithChildren> = ({ children }) => {
  const [webhooksFilterState, setWebhooksFilterState] = useState<WebhooksFilterState>({
    ...defaultWebhooksFilterState,
    ...handleHistoryToQueryParams(),
  })
  const [topics, setTopics] = useState<TopicModel[]>([])
  const { connectSession } = useReapitConnect(reapitConnectBrowserSession)
  const developerId = connectSession?.loginIdentity.developerId
  const { applicationId } = webhooksFilterState

  const [apps] = useReapitGet<Marketplace.AppSummaryModelPagedResult>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getApps],
    queryParams: { showHiddenApps: 'true', developerId, pageSize: 100 },
    fetchWhenTrue: [developerId],
  })

  const [allTopics] = useReapitGet<TopicModelPagedResult>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getWebhookTopics],
    queryParams: { showHiddenApps: 'true', developerId, pageSize: 999 },
    fetchWhenTrue: [developerId],
  })

  const [installations] = useReapitGet<Marketplace.InstallationModelPagedResult>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getInstallations],
    queryParams: { pageSize: 999, isInstalled: true, developerId, appId: applicationId },
    fetchWhenTrue: [developerId, applicationId],
  })

  const [appDetail] = useReapitGet<Marketplace.AppDetailModel>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getAppById],
    uriParams: { appId: applicationId },
    fetchWhenTrue: [applicationId],
  })

  useEffect(handleSetTopics(appDetail, allTopics, setTopics), [appDetail, allTopics])

  const webhooksDataState: WebhooksDataState = {
    apps,
    topics,
    installations,
  }

  return (
    <Provider
      value={{
        webhooksDataState,
        webhooksFilterState,
        setWebhooksFilterState,
      }}
    >
      {children}
    </Provider>
  )
}

export const useWebhooksState = (): WebhooksStateHook => {
  return useContext(WebhooksStateContext)
}
