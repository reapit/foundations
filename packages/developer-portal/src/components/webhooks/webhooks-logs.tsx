import React, { FC, useMemo } from 'react'
import { elSpan2, PersistentNotification, RowProps, Table, useSnack } from '@reapit/elements'
import dayjs from 'dayjs'
import { saveAs } from 'file-saver'
import { logger } from '@reapit/utils-react'
import { Loader } from '@reapit/elements'
import { TopicModel, WebhookLogModel } from '../../types/webhooks'
import { useWebhooksState } from './state/use-webhooks-state'
import { reapitConnectBrowserSession } from '../../core/connect-session'
import { GetActionNames, getActions, useReapitGet } from '@reapit/use-reapit-data'

export interface WebhookLogsQuery {
  from: string
  to: string
  applicationId: string
}

export interface WebhookLogsForm {
  from: Date
  to: Date
  applicationId: string
}

export const handleDownloadPayload = (payload: string, timestamp: string) => () => {
  try {
    const dataBlob = new Blob([payload], { type: 'application/json;charset=utf-8;' })
    saveAs(dataBlob, `reapit-webhook-logs-payload-${timestamp}.json`)
  } catch (err) {
    logger(err as Error)
  }
}

export const renderTopicName = (topics: TopicModel[], topicId?: string): string => {
  const subscriptionTopic = topics.find((topic) => topic.id === topicId)
  return subscriptionTopic && subscriptionTopic.name ? subscriptionTopic.name : topicId ?? ''
}

export const handleSortTableData = (logs: WebhookLogModel[], topics: TopicModel[]) => (): RowProps[] =>
  logs.map((log: WebhookLogModel) => ({
    cells: [
      {
        label: 'Timestamp',
        value: log.timeStamp ?? '',
        className: elSpan2,
        narrowTable: {
          showLabel: true,
        },
      },
      {
        label: 'URL',
        value: log.url ?? '',
        cellHasDarkText: true,
        className: elSpan2,
        narrowTable: {
          showLabel: true,
        },
      },
      {
        label: 'Topic',
        value: renderTopicName(topics, log.topicId),
        narrowTable: {
          showLabel: true,
        },
      },
      {
        label: 'Entity ID',
        value: log.entityId,
        narrowTable: {
          showLabel: true,
        },
      },
      {
        label: 'Status',
        value: String(log.statusCode),
        narrowTable: {
          showLabel: true,
        },
      },
    ],
    ctaContent: {
      icon: 'downloadSystem',
      headerContent: 'Download Payload',
      isCallToAction: true,
      onClick: handleDownloadPayload(log.payload, log.timeStamp),
    },
  }))

export const WebhooksLogs: FC = () => {
  const { webhooksFilterState, webhooksDataState } = useWebhooksState()
  const { error } = useSnack()
  const { applicationId, entityId, eventId, topicId, from, to } = webhooksFilterState
  const { topics } = webhooksDataState

  const baseQuery = {
    applicationId,
    from: dayjs(from).format('YYYY-MM-DDTHH:mm:ss'),
    to: dayjs(to).add(1, 'day').subtract(1, 'second').format('YYYY-MM-DDTHH:mm:ss'),
  }

  const entityIdQuery = entityId ? { entityId } : {}
  const eventIdQuery = eventId ? { eventId } : {}
  const topicIdQuery = topicId ? { topicId } : {}

  const queryParams = { ...baseQuery, ...entityIdQuery, ...eventIdQuery, ...topicIdQuery }

  const [logs, logsLoading] = useReapitGet<WebhookLogModel[]>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getWebhookLogs],
    queryParams,
    fetchWhenTrue: [applicationId, from, to],
    onError: (err: string) => {
      console.error(err)
      error('Something went wrong fetching webhook logs. Please adjust your query and try again')
    },
  })

  const rows = useMemo(handleSortTableData(logs ?? [], topics), [logs, topics])

  if (!applicationId || !from || !to)
    return (
      <PersistentNotification isFullWidth isExpanded intent="primary" isInline>
        No app or date range selected. Please use the filters option to get started.
      </PersistentNotification>
    )
  if (logsLoading) return <Loader />
  if (!logs?.length || !rows.length)
    return (
      <PersistentNotification isFullWidth isExpanded intent="primary" isInline>
        No logs found for this application. Select another app, date range or trigger a webhook to see the logs appear
        here.
      </PersistentNotification>
    )

  return <Table rows={rows} numberColumns={8} />
}

export default WebhooksLogs
