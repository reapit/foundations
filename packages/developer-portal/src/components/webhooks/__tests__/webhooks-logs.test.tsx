import { useReapitGet } from '@reapit/use-reapit-data'
import React from 'react'
import { render } from '../../../tests/react-testing'
import { mockWebhookLogs } from '../../../tests/__stubs__/webhooks'
import { useWebhooksState } from '../state/use-webhooks-state'
import { mockWebhooksState } from '../state/__mocks__/use-webhooks-state'
import { WebhooksLogs } from '../webhooks-logs'

jest.mock('../state/use-webhooks-state')

jest.mock('@reapit/use-reapit-data', () => ({
  ...jest.requireActual('@reapit/use-reapit-data'),
  useReapitGet: jest.fn(() => [mockWebhookLogs, false, undefined, jest.fn()]),
}))

const mockUseReapitGet = useReapitGet as jest.Mock
const mockUseWebhooksState = useWebhooksState as jest.Mock

describe('WebhooksLogs', () => {
  it('should match a snapshot where there are subscriptions', () => {
    mockUseWebhooksState.mockReturnValue({
      ...mockWebhooksState,
      webhooksFilterState: {
        ...mockWebhooksState.webhooksFilterState,
        applicationId: 'MOCK_ID',
      },
    })
    expect(render(<WebhooksLogs />)).toMatchSnapshot()
  })

  it('should match a snapshot where loading', () => {
    mockUseWebhooksState.mockReturnValue({
      ...mockWebhooksState,
      webhooksFilterState: {
        ...mockWebhooksState.webhooksFilterState,
        applicationId: 'MOCK_ID',
      },
    })
    mockUseReapitGet.mockReturnValue([null, true, undefined, jest.fn()])
    expect(render(<WebhooksLogs />)).toMatchSnapshot()
  })

  it('should match a snapshot where no data', () => {
    mockUseWebhooksState.mockReturnValue({
      ...mockWebhooksState,
      webhooksFilterState: {
        ...mockWebhooksState.webhooksFilterState,
        applicationId: 'MOCK_ID',
      },
    })
    mockUseReapitGet.mockReturnValue([[], false, undefined, jest.fn()])
    expect(render(<WebhooksLogs />)).toMatchSnapshot()
  })

  it('should match a snapshot where no appId', () => {
    mockUseWebhooksState.mockReturnValue(mockWebhooksState)
    mockUseReapitGet.mockReturnValue([null, false, undefined, jest.fn()])
    expect(render(<WebhooksLogs />)).toMatchSnapshot()
  })
})
