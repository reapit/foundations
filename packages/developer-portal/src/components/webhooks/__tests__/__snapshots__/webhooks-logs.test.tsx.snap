// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebhooksLogs should match a snapshot where loading 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <mock-styled.div />
      <mock-styled.div
        classname=""
      >
        <mock-styled.div>
          <mock-styled.div />
        </mock-styled.div>
      </mock-styled.div>
    </div>
  </body>,
  "container": <div>
    <mock-styled.div />
    <mock-styled.div
      classname=""
    >
      <mock-styled.div>
        <mock-styled.div />
      </mock-styled.div>
    </mock-styled.div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WebhooksLogs should match a snapshot where no appId 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <mock-styled.div />
      <mock-styled.div
        classname="mock-css. mock-css. mock-css. mock-css."
      >
        <div
          class="mock-css.
  padding: 0 1rem;
  display: flex;
  align-items: center;
  border-radius: var(--default-border-radius) 0 0 var(--default-border-radius);
  cursor: pointer;

  , {
    color: var(--color-white);
  }
"
          data-testid="close-icon"
        >
          <mock-styled.span
            classname=""
          >
            <svg />
          </mock-styled.span>
        </div>
        <div
          class="mock-css.
  padding: 1rem;
  opacity: 0;
  transition: 0.5s;
  width: 100%;
"
        >
          No app or date range selected. Please use the filters option to get started.
        </div>
      </mock-styled.div>
    </div>
  </body>,
  "container": <div>
    <mock-styled.div />
    <mock-styled.div
      classname="mock-css. mock-css. mock-css. mock-css."
    >
      <div
        class="mock-css.
  padding: 0 1rem;
  display: flex;
  align-items: center;
  border-radius: var(--default-border-radius) 0 0 var(--default-border-radius);
  cursor: pointer;

  , {
    color: var(--color-white);
  }
"
        data-testid="close-icon"
      >
        <mock-styled.span
          classname=""
        >
          <svg />
        </mock-styled.span>
      </div>
      <div
        class="mock-css.
  padding: 1rem;
  opacity: 0;
  transition: 0.5s;
  width: 100%;
"
      >
        No app or date range selected. Please use the filters option to get started.
      </div>
    </mock-styled.div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WebhooksLogs should match a snapshot where no data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <mock-styled.div />
      <mock-styled.div
        classname="mock-css. mock-css. mock-css. mock-css."
      >
        <div
          class="mock-css.
  padding: 0 1rem;
  display: flex;
  align-items: center;
  border-radius: var(--default-border-radius) 0 0 var(--default-border-radius);
  cursor: pointer;

  , {
    color: var(--color-white);
  }
"
          data-testid="close-icon"
        >
          <mock-styled.span
            classname=""
          >
            <svg />
          </mock-styled.span>
        </div>
        <div
          class="mock-css.
  padding: 1rem;
  opacity: 0;
  transition: 0.5s;
  width: 100%;
"
        >
          No logs found for this application. Select another app, date range or trigger a webhook to see the logs appear here.
        </div>
      </mock-styled.div>
    </div>
  </body>,
  "container": <div>
    <mock-styled.div />
    <mock-styled.div
      classname="mock-css. mock-css. mock-css. mock-css."
    >
      <div
        class="mock-css.
  padding: 0 1rem;
  display: flex;
  align-items: center;
  border-radius: var(--default-border-radius) 0 0 var(--default-border-radius);
  cursor: pointer;

  , {
    color: var(--color-white);
  }
"
        data-testid="close-icon"
      >
        <mock-styled.span
          classname=""
        >
          <svg />
        </mock-styled.span>
      </div>
      <div
        class="mock-css.
  padding: 1rem;
  opacity: 0;
  transition: 0.5s;
  width: 100%;
"
      >
        No logs found for this application. Select another app, date range or trigger a webhook to see the logs appear here.
      </div>
    </mock-styled.div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WebhooksLogs should match a snapshot where there are subscriptions 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <mock-styled.div />
      <mock-styled.div
        data-has-call-to-action="true"
        data-has-expandable-action="false"
        data-num-columns-excl-action-col="7"
      >
        <mock-styled.div>
          <mock-styled.div
            classname="mock-css.
  grid-column-end: span 2;
"
          >
            Timestamp
          </mock-styled.div>
          <mock-styled.div
            classname="mock-css.
  grid-column-end: span 2;
"
          >
            URL
          </mock-styled.div>
          <mock-styled.div>
            Topic
          </mock-styled.div>
          <mock-styled.div>
            Entity ID
          </mock-styled.div>
          <mock-styled.div>
            Status
          </mock-styled.div>
          <mock-styled.div>
            Download Payload
          </mock-styled.div>
        </mock-styled.div>
        <mock-styled.div
          classname=""
        >
          <mock-styled.div>
            <mock-styled.div
              classname="mock-css.
  grid-column-end: span 2;
"
            >
              <mock-styled.div
                data-narrow-label="Timestamp"
              >
                2021-04-13T12:17:04.177+00:00
              </mock-styled.div>
            </mock-styled.div>
            <mock-styled.div
              classname="mock-css.
  grid-column-end: span 2;
 mock-css."
            >
              <mock-styled.div
                data-narrow-label="URL"
              >
                https://3391b00a1f9c1fcae3e09d28dd99b7c9.m.pipedream.net
              </mock-styled.div>
            </mock-styled.div>
            <mock-styled.div
              classname=""
            >
              <mock-styled.div
                data-narrow-label="Topic"
              >
                application.install
              </mock-styled.div>
            </mock-styled.div>
            <mock-styled.div
              classname=""
            >
              <mock-styled.div
                data-narrow-label="Entity ID"
              >
                MOCK_ID
              </mock-styled.div>
            </mock-styled.div>
            <mock-styled.div
              classname=""
            >
              <mock-styled.div
                data-narrow-label="Status"
              >
                200
              </mock-styled.div>
            </mock-styled.div>
            <mock-styled.div>
              <mock-styled.div>
                <mock-styled.span
                  classname="mock-css."
                  style="font-size: 1.2rem;"
                >
                  <svg
                    style="font-size: 1.2rem;"
                  />
                </mock-styled.span>
              </mock-styled.div>
            </mock-styled.div>
          </mock-styled.div>
        </mock-styled.div>
      </mock-styled.div>
    </div>
  </body>,
  "container": <div>
    <mock-styled.div />
    <mock-styled.div
      data-has-call-to-action="true"
      data-has-expandable-action="false"
      data-num-columns-excl-action-col="7"
    >
      <mock-styled.div>
        <mock-styled.div
          classname="mock-css.
  grid-column-end: span 2;
"
        >
          Timestamp
        </mock-styled.div>
        <mock-styled.div
          classname="mock-css.
  grid-column-end: span 2;
"
        >
          URL
        </mock-styled.div>
        <mock-styled.div>
          Topic
        </mock-styled.div>
        <mock-styled.div>
          Entity ID
        </mock-styled.div>
        <mock-styled.div>
          Status
        </mock-styled.div>
        <mock-styled.div>
          Download Payload
        </mock-styled.div>
      </mock-styled.div>
      <mock-styled.div
        classname=""
      >
        <mock-styled.div>
          <mock-styled.div
            classname="mock-css.
  grid-column-end: span 2;
"
          >
            <mock-styled.div
              data-narrow-label="Timestamp"
            >
              2021-04-13T12:17:04.177+00:00
            </mock-styled.div>
          </mock-styled.div>
          <mock-styled.div
            classname="mock-css.
  grid-column-end: span 2;
 mock-css."
          >
            <mock-styled.div
              data-narrow-label="URL"
            >
              https://3391b00a1f9c1fcae3e09d28dd99b7c9.m.pipedream.net
            </mock-styled.div>
          </mock-styled.div>
          <mock-styled.div
            classname=""
          >
            <mock-styled.div
              data-narrow-label="Topic"
            >
              application.install
            </mock-styled.div>
          </mock-styled.div>
          <mock-styled.div
            classname=""
          >
            <mock-styled.div
              data-narrow-label="Entity ID"
            >
              MOCK_ID
            </mock-styled.div>
          </mock-styled.div>
          <mock-styled.div
            classname=""
          >
            <mock-styled.div
              data-narrow-label="Status"
            >
              200
            </mock-styled.div>
          </mock-styled.div>
          <mock-styled.div>
            <mock-styled.div>
              <mock-styled.span
                classname="mock-css."
                style="font-size: 1.2rem;"
              >
                <svg
                  style="font-size: 1.2rem;"
                />
              </mock-styled.span>
            </mock-styled.div>
          </mock-styled.div>
        </mock-styled.div>
      </mock-styled.div>
    </mock-styled.div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
